code-examples/low-latency-sessions/main.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-redis/redis/v8"
+
+	client "github.com/ory/kratos-client-go"
+)
+
+type kratosMiddleware struct {
+	client *client.APIClient
+	redis  *redis.Client
+}
+
+func NewMiddleware() *kratosMiddleware {
+	configuration := client.NewConfiguration()
+	configuration.Servers = []client.ServerConfiguration{
+		{
+			URL: "https://playground.projects.oryapis.com",
+		},
+	}
+	return &kratosMiddleware{
+		client: client.NewAPIClient(configuration),
+		redis: redis.NewClient(&redis.Options{
```

Comment:
I think this is really cool to use redis! An alternative, which would probably be a little bit faster, would be to use something like:

- https://github.com/hashicorp/golang-lru
- https://dgraph.io/blog/post/caching-in-go/

That would make the tutorial a bit easier because you don't need to set up redis.

However, redis is of course better when in production. What do you think?

Reply:
golang-lru, goleveldb and other in-memory embeddable solutions are great to have in the application. However, for production usage, people use Redis or Memcached. I'll add an example for in-memory caches but I'll add a couple of notes, like this does not scale well and you need to use redis or something similar in production.

Reply:
That's a really good idea, thanks :)

Reply:
I added an example of using ristretto cache. golang-lru does not support TTL out of the box.

Reply:
Even better :)